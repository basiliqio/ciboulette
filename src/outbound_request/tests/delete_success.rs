use super::*;

#[test]
fn simple_update() {
    let store = gen_bag();
    let url = Url::parse("http://localhost/").unwrap();
    let opt = url::Url::options().base_url(Some(&url));
    const URL: &str = "/comments/073b5936-0acb-4601-b4b7-9de607dfc2ef";
    const INTENTION: CibouletteIntention = CibouletteIntention::Delete;

    let parsed_url = opt.parse(URL).unwrap();
    let builder = CibouletteInboundRequestBuilder::new(INTENTION, &parsed_url, &None);
    let request = builder.build(&store).unwrap();
    let res = CibouletteDeleteRequest::try_from(request).unwrap();
    let response = CibouletteOutboundRequestDataBuilder::<'_, '_, String, _>::new(&res, vec![])
        .build(&store)
        .unwrap();
    assert_eq!(response.status(), CibouletteResponseStatus::OkEmpty);
    assert_json_snapshot!(response);
}
