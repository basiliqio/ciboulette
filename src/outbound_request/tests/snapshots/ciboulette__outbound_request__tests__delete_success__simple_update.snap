---
source: src/outbound_request/tests/delete_success.rs
expression: response

---
{
  "jsonapi": {
    "version": "1.0"
  },
  "data": null
}
